### PR TITLE
Write handler for prom remote, Fix context propagation

### PR DESCRIPTION
**Pull request by nikunjgit**

### CODE REVIEW

File: services/m3coordinator/handler/promRead.go [hunk 1]
```diff
@@ -21,12 +23,14 @@ func NewPromReadHandler() http.Handler {
...
 
 func (h *PromReadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	_, err := h.parseRequest(w, r)
+	req, err := h.parseRequest(w, r)
 	if err != nil {
 		return
 	}
 
-	// TODO: Actual read
+	// TODO: Actual read instead of logging
```
Comment (benraskin92) [inline services/m3coordinator/handler/promRead.go]: `TODO` => `TODO (nikunj)` - just so we can keep track of who's responsible for what

File: services/m3coordinator/main/main.go [hunk 2]
```diff
@@ -21,7 +36,36 @@ func main() {
...
+
+	a.HelpFlag.Short('h')
+
+	a.Flag("config.file", "M3Coordinator configuration file path.").
+		Default("coordinator.yml").StringVar(&cfg.configFile)
+
+	a.Flag("query.port", "Address to listen on.").
+		Default("0.0.0.0:4116").StringVar(&cfg.listenAddress)
+
+	a.Flag("query.timeout", "Maximum time a query may take before being aborted.").
```
Comment (benraskin92) [inline services/m3coordinator/main/main.go]: Is this just a placeholder?

File: services/m3coordinator/main/main.go [hunk 3]
```diff
@@ -21,7 +36,36 @@ func main() {
...
+		Default("2m").DurationVar(&cfg.queryTimeout)
+
+	a.Flag("query.max-concurrency", "Maximum number of queries executed concurrently.").
```
Comment (benraskin92) [inline services/m3coordinator/main/main.go]: Same here

File: services/m3coordinator/main/main.go [hunk 4]
```diff
@@ -3,14 +3,29 @@ package main
...
 	"net/http"
 	"os"
+	"path/filepath"
+	"time"
 
 	"github.com/m3db/m3coordinator/services/m3coordinator/httpd"
 	"github.com/m3db/m3coordinator/util/logging"
+
+	"go.uber.org/zap"
+	"gopkg.in/alecthomas/kingpin.v2"
```
Comment (benraskin92) [inline services/m3coordinator/main/main.go]: Out of curiosity, why not go with [Cobra](https://github.com/spf13/cobra) instead?

Comment (nikunjgit) [reply services/m3coordinator/main/main.go]: no big reason, we also use other packages from alecthomas like gometalinter and prometheus also uses kingpin

File: services/m3coordinator/handler/promWrite.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+	if err != nil {
+		return
+	}
+
+	// TODO: Actual write instead of logging
+	logging.WithContext(r.Context()).Info("Write request", zap.Any("req", req))
+}
+func (h *PromWriteHandler) parseRequest(w http.ResponseWriter, r *http.Request) (*prompb.WriteRequest, error) {
+	if r.Body == nil {
+		err := fmt.Errorf("empty request body")
```
[the remark this answers is not in the document]
Comment (nikunjgit) [reply services/m3coordinator/handler/promWrite.go]: refactored

File: services/m3coordinator/handler/promRead.go (hunk 1 above)
Comment (nikunjgit) [reply services/m3coordinator/handler/promRead.go]: done

File: services/m3coordinator/main/main.go (hunk 2 above)
Comment (nikunjgit) [reply services/m3coordinator/main/main.go]: yes

File: services/m3coordinator/main/main.go (hunk 3 above)
Comment (nikunjgit) [reply services/m3coordinator/main/main.go]: yes

**Pull request merged**